### PR TITLE
banshee: Support for mulhsu/mulhu/mul instructions

### PR DESCRIPTION
**Pull request by noah95**
This adds support for the mulhu/mul instructions. afaik LLVM does not distinguish between signed/unsigned multiplication (see https://releases.llvm.org/11.0.0/docs/LangRef.html#mul-instruction), but perhaps someone can verify that.

### CODE REVIEW

**Review by paulsc96 — APPROVED** [review]
I am unsure about the sign handling, too, but if this implementation works in your application, it is probably most productive to merge this now and fix it later if issues arise.

**Comment by noah95** [comment]
Sorry what i implemented is complete rubbish and doesn't at all execute a `mulh`, do not merge! Will fix it and also do a proper test

**Comment by noah95** [comment]
Ok, it should be correct now. There is also a different in signed/unsigned during casing
```
x5:80000003 x6:00000002 x28=00000001    # mulhu   t3, t0, t1
x5:deadbeef x6:deadbeef x28=c1b1cd12    # mulhu   t3, t0, t1
x5:ffffffff x6:ffffffff x28=fffffffe    # mulhu   t3, t0, t1
x5:ffffffff x6:ffffffff x28=00000000    # mulh    t3, t0, t1
x5:80000000 x6:00000002 x28=ffffffff    # mulh    t3, t0, t1
x5:deadbeef x6:deadbeef x28=04564f34    # mulh    t3, t0, t1
```

File: sw/banshee/src/tran.rs
Comment (SamuelRiedel) [inline sw/banshee/src/tran.rs:2151]: Nice. Could we add `mulhsu` as well?
```suggestion
            }
            riscv::OpcodeRdRs1Rs2::Mulhsu => {
                let tmp = LLVMBuildMul(
                    self.builder,
                    LLVMBuildSExt(self.builder, rs1, LLVMInt64Type(), NONAME),
                    LLVMBuildZExt(self.builder, rs2, LLVMInt64Type(), NONAME),
                    name,
                );
                let tmp = LLVMBuildLShr(
                    self.builder,
                    tmp,
                    LLVMConstInt(LLVMInt64Type(), 32 as u64, 0),
                    NONAME,
                );
                LLVMBuildTrunc(self.builder, tmp, LLVMInt32Type(), NONAME)
            }
```

Comment (noah95) [reply sw/banshee/src/tran.rs:2151]: could you also write a test for it?

**Review by SamuelRiedel — APPROVED** [review]

**Pull request merged**
